src/txn/txn_timestamp.c

```diff
@@ -699,19 +734,17 @@ void
 __wt_txn_clear_commit_timestamp(WT_SESSION_IMPL *session)
 {
 	WT_TXN *txn;
-	WT_TXN_GLOBAL *txn_global;
+	uint32_t flags;
 
 	txn = &session->txn;
-	txn_global = &S2C(session)->txn_global;
 
 	if (!F_ISSET(txn, WT_TXN_PUBLIC_TS_COMMIT))
 		return;
+	flags = txn->flags;
+	LF_CLR(WT_TXN_PUBLIC_TS_COMMIT);
 
-	__wt_writelock(session, &txn_global->commit_timestamp_rwlock);
-	TAILQ_REMOVE(&txn_global->commit_timestamph, txn, commit_timestampq);
-	--txn_global->commit_timestampq_len;
-	__wt_writeunlock(session, &txn_global->commit_timestamp_rwlock);
-	F_CLR(txn, WT_TXN_PUBLIC_TS_COMMIT);
+	WT_PUBLISH(txn->clear_ts_queue, true);
```

Comment:
I think a comment above here would be useful. Something like:
Notify other operations that our transaction is inactive, and can be safely cleaned up from the queue. It's not necessary for the entry to be removed prior to returning since there is no guarantee around the order of commit completing.

Reply:
Fixed. Thanks.